src/Resources/Adjustments/Operations/CreateAdjustment.php

```diff
@@ -22,21 +22,39 @@ class CreateAdjustment implements \JsonSerializable
      */
     public function __construct(
         public readonly Action $action,
-        public readonly array|Undefined $items,
+        public readonly array|Undefined|null $items,
         public readonly string $reason,
         public readonly string $transactionId,
         public readonly AdjustmentType|Undefined $type = new Undefined(),
     ) {
-        if ($this->type === AdjustmentType::Partial() && ($this->items instanceof Undefined || empty($this->items))) {
+        $typeIsFull = AdjustmentType::Full()->equals($this->type);
```

Comment:
`$this->type === AdjustmentType::Partial()` strict comparison can't be used here